docs/java_keystore.md

```diff
@@ -0,0 +1,60 @@
+# Java PKCS11 Keystore
+
+A Java keystore is a repository where symmetric, asymmetric and certificates are stored.
+In the context of SSL/TLS, the server private key and its associated certificate chain can
+be imported in the keystore by using the PKCS11 storetype via the configured PKCS#11 provider.
+Details can be found in the [Java PKCS#11 Reference Guide](https://docs.oracle.com/en/java/javase/17/security/pkcs11-reference-guide1.html#GUID-30E98B63-4910-40A1-A6DD-663EAF466991)
+which uses the SunPKCS11 provider on an AWS EC2 instance.
+
+In this implementation, the PKCS#11 provider is intended to be executed within the confines of a Nitro Enclave.
+
+> NOTE: In the current implementation, only the SunPKCS11 provider keystore Read-Only requirements are supported. See [here](https://docs.oracle.com/en/java/javase/17/security/pkcs11-reference-guide1.html#GUID-F068390B-EB41-48A0-A713-B4CBCC72285D). This is because the enclave pkcs#11 token is write-protected and
+> does not allow creating, modifying or destroying cryptographic objects after provisioning.
+
+1. Install a Java Development Kit of your choice. For example, [Amazon Corretto](https://docs.aws.amazon.com/corretto/).
+
+2. Create the SunPKCS11 provider configuration file in your location of choice
+```sh
+cat /etc/pkcs11/keystore.conf
+
+name = p11ne
+description = "PKCS#11 Keystore"
+library = /usr/lib64/libp11-kit.so.0
+slotListIndex = 0
```

Comment:
Is this effectively hardcoding the Java Provider to target slot 0 of the token? Or the first token to be enumerated of a set of tokens?

Reply:
Yes, it's the first in `C_GetSlotList()` since we always provision implicitly from 0 as the current provisioning API does not have explicit slot ID (index) support from instance-side. It already defaults to this. I will remove it to avoid confusion as these parameters are already documented in the Sun PKCS#11 guide.